packages/gatsby-theme/src/components/Share/ShareLinkedin.js

```diff
@@ -0,0 +1,35 @@
+import PropTypes from "prop-types"
+import { LinkedinShareButton } from "react-share"
+import Icon from "../Icon"
+
+const ShareLinkedin = ({ title, to, summary, source, children, ...props }) => {
+  return (
+    <LinkedinShareButton
+      url={to}
+      title={title}
+      summary={summary}
+      source={source}
+      resetButtonStyle={false}
+      {...props}
+    >
+      <Icon icon='linkedin' />
+      {children}
+    </LinkedinShareButton>
+  )
+}
+
```

Comment:
Same refactoring of using a single component with a Hashmap to resolve the specific component from the "react-share" library